app/src/components/Shipment/OrderCard.svelte

```diff
@@ -1,38 +1,93 @@
 <script lang="ts">
-	import type { GeoLocation } from '$src/utils/idl/shipment';
+	import type {
+		Shipment,
+		ShipmentDimensions,
+		GeoLocation,
+		Geography,
+		ShipmentDetails
+	} from '$src/utils/idl/shipment';
+	import CancelConfirmModal from './CancelConfirmModal.svelte';
 	import Card from './Card.svelte';
 
-	export let name: string;
-	export let from: GeoLocation;
-	export let to: GeoLocation;
-	export let date: Date;
+	type Entries<T> = {
+		[K in keyof T]: [K, T[K]];
+	}[keyof T][];
+
+	export let shipmentData: Shipment<string, number, string>;
+
+	$: dimensions = Object.entries(shipmentData.shipment.dimensions) as Entries<ShipmentDimensions>;
+	$: locations = Object.entries(shipmentData.shipment.geography) as Entries<Geography>;
+	$: properties = Object.entries(shipmentData.shipment.details) as Entries<ShipmentDetails>;
+
+	let isBuyClicked: boolean = false;
+
+	function handleBuyOrder(e: Event) {
+		console.log(e);
+	}
 </script>
 
 <Card>
-	<h3 slot="name">
-		{name}
-	</h3>
-	<p slot="image">There will be image</p>
+	<h3 slot="name">name</h3>
+	<p slot="image">image</p>
 	<p slot="first-info">info 1</p>
 	<p slot="second-info">info 2</p>
 	<svelte:fragment slot="details">
 		<details>
 			<summary>Precise location</summary>
 			<ul>
-				<li>Source: {Object.values(from)}</li>
-				<li>Destination: {Object.values(to)}</li>
+				<li>Source: {0}</li>
```

Comment:
Why is it hardcoded here

Reply:
It's for listing so does not matter, still thinking how to layout order

Reply:
i've displayed some in https://github.com/tommysr/trans-protocol/pull/11/commits/5a8c538fa48e3681dad71dadd4c028fc410cac74